File: problems/3021_alice_and_bob_playing_flower_game/3021_alice_and_bob_playing_flower_game.py
class Solution:
    def flowerGame(self, n: int, m: int) -> int:
        xOdd = (n + 1) // 2
        yOdd = (m + 1) // 2
        xEven = n // 2
        yEven = m // 2
        return xEven * yOdd + xOdd * yEven

File: problems/3021_alice_and_bob_playing_flower_game/3021_alice_and_bob_playing_flower_game.java
class Solution {
    public long flowerGame(int n, int m) {
        long xOdd = (n + 1) / 2;
        long yOdd = (m + 1) / 2;
        long xEven = n / 2;
        long yEven = m / 2;
        return xOdd * yEven + xEven * yOdd;
    }
}
